pandas/core/arrays/categorical.py

```diff
@@ -2015,7 +2018,20 @@ def __repr__(self) -> str:
 
     # ------------------------------------------------------------------
 
-    def __getitem__(self, key):
+    @overload
+    def __getitem__(self, key: int | np.integer) -> object:
+        ...
+
+    @overload
+    def __getitem__(
+        self: CategoricalT,
+        key: slice | np.ndarray | list[int] | PositionalIndexerTuple,
+    ) -> CategoricalT:
+        ...
+
+    def __getitem__(
+        self: CategoricalT, key: PositionalIndexer2D
+    ) -> CategoricalT | object:
```

Comment:
object-> Any?

Reply:
Can a categorical be based on a List?

Reply:
im not clear what "based on" means here

Reply:
If we use `Any`, then that means any type and mypy stops further type checking.  From some other PR, the idea here is that we can return an `object`, which means other type checking would work downstream.

My comment about "categorical be based on a List" was misleading.

Reply:
for consistency with other places where I did `__getitem__` overloads, I changed this to `Any`